### PR TITLE
Run code-ql workflow only when certain files change

### PR DESCRIPTION
**Pull request by Flyingmana**
run code-ql only when certain files change

based on parts of #3773

### CODE REVIEW

**Comment by sreichel** [comment]
Nice.

Please review this too ... there is no v20 branch ...

```
  push:
    branches: [ "main", "v20", "v19" ]
```

**Comment by sreichel** [comment]
When you touch this file you can also fix this, Just remove these lines shoulb be okay.

**Comment by sreichel** [comment]
> thats something for a different PR

Everytime, everything I say ... I only hear no, not, but ...

... but adopt my try ...  :angry:

**Comment by sreichel** [comment]
Das nächste mal, sag Bescheid! Das Du meine Arbeit kopierst möchte ich auf keinstn Fall!

**Comment by Flyingmana** [comment]
> Das nächste mal, sag Bescheid! Das Du meine Arbeit kopierst, möchte ich **auf keinstn Fall!**

technically, you provide the code under the license of the project you create the Pull Request for, regardless of if it gets merged there or somewhere else.

I wrote directly in the beginning the changes are based on your PR, which you closed for impatience.

This are also 2 more personal attacks on me today, so we solve this for the future by you getting a permanent ban from the project.

**Comment by Flyingmana** [comment]
the PR of https://github.com/OpenMage/magento-lts/pull/3794 should contain the last part needed to make the unit tests green again

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@
 name: "CodeQL"
 
 on:
+  workflow_call:
+  workflow_dispatch:
   push:
     branches: [ "main", "v20", "v19" ]
```
Comment (fballiano) [inline .github/workflows/codeql-analysis.yml:18]:
```suggestion
    branches: [ "main", "next", "v19" ]
```

Comment (ADDISON74) [reply .github/workflows/codeql-analysis.yml:18]: A good observation.

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**